### PR TITLE
Feat/saved notes features (Trash, Archive, Edit)

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added edit, trash and archive features

- Added note can be edited in main notes page and saved
- ARCHIVE FEATURES
> - Note can be archived (Send to archive page)
> - Note can be restored from archive 
> - Delete from archive will delete note permanently (Not add to Trash as mockbee doesn't have)

- TRASH FEATURES
> - Note can be deleted from main page and it gets added to Trash Page (No TrashController in mockbee so did this through array)
> - Deleted note gets removed from saved notes and added to trash (from trash it cannot be recovered)

### What not to check

- **Features which mockbee doesn't have which needs backend code to be added**
- Change colour
- Date added
- Pinning unpinning
- Labels

trash-archive

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ function App() {
             <Route path="/login" element={<LoginPage />} />
```
Comment (sakshi006) [inline src/App.js:16]: Add a "not found" path as well

File: src/pages/ArchivePage/ArchivePage.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { useLocation } from "react-router-dom";
+import { useNote } from "../../contexts";
+import { NavSide, NoteCard } from "../../components";
+import "../NotesPage/notes-page.css";
+
+const ArchivePage = () => {
+  const {
+    noteState: { notesArchive },
```
Comment (sakshi006) [inline src/pages/ArchivePage/ArchivePage.jsx:8]: Good use of destructuring

**Review by sakshi006 — APPROVED** [review]
Looks good. 
The code is clean and UI is amazing.
Just a suggestion: Maybe you can add a count of notes that are created as well as that are in the archive tab.

**Review by krgarima — APPROVED** [review]
Clean UI and great folder structure. App is also working smoothly.
(Optional suggestion): You can add a note recovery feature for trashed notes.

File: src/utilities/trashNote.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      {
+        headers: Headers,
+      }
+    );
+    if (status >= 200 && status <= 300) {
+      noteDispatch({ type: "ADD_NOTE", payload: notes });
+      noteDispatch({ type: "TRASH_NOTE", payload: note });
+    }
+  } catch (error) {
+    console.log(error.message);
```
Comment (krgarima) [inline src/utilities/trashNote.js:21]: Don't leave console.log in code. Add a toast here which shows error message.

Comment (bhavyaj12) [reply src/utilities/trashNote.js:21]: Yes will add later, as enhancement after this submission

**Pull request merged**